### ISSUE TITLE
API-35: MicroProfile Rest Client 1.2 in OpenLiberty

### ISSUE THREAD

**Issue by andymc12** [OpenLiberty/open-liberty#5941]
Implement MicroProfile Rest Client 1.2.

[Main MP Rest Client 1.2 GitHub page](https://github.com/eclipse/microprofile-rest-client)

[Latest milestone release](https://github.com/eclipse/microprofile-rest-client/releases/tag/1.2-m1)

[What's New in 1.2? (release notes)](https://github.com/eclipse/microprofile-rest-client/blob/master/spec/src/main/asciidoc/release_notes.asciidoc)

WebSphere Analysis Document - pending approval.

**Comment by andymc12** [comment]
[Aha! Feature API-35](https://bigblue.aha.io/features/API-35)

**Comment by marikaj123** [comment]
Does the feature require to update any nls messages?

**Comment by andymc12** [comment]
@marikaj123
> Does the feature require to update any nls messages?

No, all of the NLS messages is part of the Apache CXF open source project.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?

**Comment by andymc12** [comment]
> Serviceability Approval Comment - Please answer the following questions for serviceability approval:
> 
> WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR? Have these issues been addressed in the implementation?

Yes

> Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).
> a) What problem paths were tested and demonstrated?

Debugging invalid paths or client errors.

> b) Who did you demo to?

L2 - note that the _live_ demo was done for previous versions of the Rest Client.  For 1.2, we only presented this in static charts.

> c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

Yes

> SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them. Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
> a) Who conducted SVT tests for this feature?

Brian Hanczaryk

> b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?

TBD

> Which L2 / L3 queues will handle PMRs for this feature? Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it. Ask Don Bourne if you need links or more info.

L2: WASWEB
L3: WL3JRS

**Comment by bsbyrd1** [comment]
Giving ID approval, noting that I had a chat with Andy, who is working with YK to update the guides for this feature.

**Comment by andymc12** [comment]
Shipped in 19.0.0.3
